### PR TITLE
#215 | Resolve_Node_Duplication_in_db_seed_Command

### PR DESCRIPTION
**Pull request by adeel-ahmad-awan**
Ticket

- [Resolve Node Duplication in `db:seed` Command#215](https://app.zenhub.com/workspaces/th-developmentdesigning-board-6565ea6f792dae0625b5c739/issues/gh/scilifelab-traininghub-platform/tess/215)

### CODE REVIEW

**Review by umunixie — COMMENTED** [review]
After looking at it for some time, I can't say I understand what's going on.
In particular, I don't understand

1. Why were we using a task instead of `db/seeds.rb`?
2. Why generate constants in runtime?
3. Why have the Scilife constant at all, it doesn't seem to be used anywhere? The External constant is only used in one place as far as I could find, for deciding which option should be checked in a radio button group. Is that necessary?

File: app/models/node.rb [hunk 1]
```diff
@@ -60,6 +54,57 @@ class Node < ApplicationRecord
...
+  #       "slug": "external"
+  #     }
+  #   ]
+  # }
+  def self.load_constants
+    data = JSON.parse(File.read(NODE_FILE_PATH))
+    data['nodes'].each do |node|
+      # Define constants dynamically
+      const_name = node['constant_name']
+      next unless const_name # Skip if constant_name is missing
```
Comment (umunixie) [inline app/models/node.rb]: Silently swallowing errors scares me.

**Comment by adeel-ahmad-awan** [comment]
> After looking at it for some time, I can't say I understand what's going on. In particular, I don't understand
> 
>     1. Why were we using a task instead of `db/seeds.rb`?
> 
>     2. Why generate constants in runtime?
> 
>     3. Why have the Scilife constant at all, it doesn't seem to be used anywhere? The External constant is only used in one place as far as I could find, for deciding which option should be checked in a radio button group. Is that necessary?

@umunixie 

**Why were we using a task instead of `db/seeds.rb`?**
Previously, we opted for a straightforward approach by creating a custom command to handle node population, as we had limited knowledge of the overall system at that time. However, during this PR, we encountered issues related to the seeding of nodes and identified opportunities to improve the process. As a result, we refactored the code to align with the use of db/seeds.rb.

**Why generate constants in runtime?**
To avoid duplicating logic and hardcoding values, we dynamically handle constants at runtime. While these are generated during initialization, they are created only once and remain accessible throughout the application’s lifecycle, ensuring maintainability.

**Why have the Scilife constant at all, it doesn't seem to be used anywhere? The External constant is only used in one place as far as I could find, for deciding which option should be checked in a radio button group. Is that necessary?**
This was done for future proofing the code. 


-----

**With that being said**, if you feel this approach is overengineered, I can revert to the previous implementation of hardcoded constants.

**Comment by umunixie** [comment]
I wouldn't say it's overengineering, but I think it's risky to define constants in runtime in particular. Constants are meant to be this reliable thing which is always there for you, not something that can fail. I would strongly advise against generating constants at runtime.

I would also advise against having unused variables/constants/whatever in the code, since it's only creates confusion and potential for bugs.

**Review by umunixie — APPROVED** [review]

**Review by harshita3199 — APPROVED** [review]
Looks good to me!! Tested in dev and pre-prod, Able to seed the data

**Pull request merged**